server/container_create.go

```diff
@@ -99,64 +99,45 @@ func addDevices(sb *sandbox, containerConfig *pb.ContainerConfig, specgen *gener
 
 // buildOCIProcessArgs build an OCI compatible process arguments slice.
 func buildOCIProcessArgs(containerKubeConfig *pb.ContainerConfig, imageOCIConfig *v1.Image) ([]string, error) {
-	processArgs := []string{}
-	var processEntryPoint, processCmd []string
+	//# Start the nginx container using the default command, but use custom
+	//arguments (arg1 .. argN) for that command.
+	//kubectl run nginx --image=nginx -- <arg1> <arg2> ... <argN>
+
+	//# Start the nginx container using a different command and custom arguments.
+	//kubectl run nginx --image=nginx --command -- <cmd> <arg1> ... <argN>
 
 	kubeCommands := containerKubeConfig.Command
 	kubeArgs := containerKubeConfig.Args
 
-	if imageOCIConfig == nil {
```

Comment:
> Adding the check inside the function will make the code more readable...I undestand we want to be bug compatible with what Docker does, it does not mean our implementation can't be nicer ;)

I think I don't follow anymore :smile: We're doing this:

1. get the image config w/o checking for `nil` here https://github.com/kubernetes-incubator/cri-o/pull/572/files#diff-dba9598e19f7903533dc9add98d410d2R580 (I just added the check but right now we're not checking for nilness!)
2. w/o my nil check, if the image config is nil we instanly panic here https://github.com/kubernetes-incubator/cri-o/pull/572/files#diff-dba9598e19f7903533dc9add98d410d2R585 (This is a bug we do have right now in master!!!)
3. we check image config nilness in https://github.com/kubernetes-incubator/cri-o/pull/572/files#diff-dba9598e19f7903533dc9add98d410d2R598 but it's far too late since other stuff has already tried to dereference a possible nil pointer panicking the whole server

I mean, we can still keep the check in the function, sure...